src/base/string_buffer.hpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2008 Search Solution Corporation. All rights reserved by Search Solution.
+ *
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 2 of the License, or
+ *   (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+#pragma once
+#ifdef __linux__
+#include <stddef.h>//size_t on Linux
```

Comment:
#include <stddef.h> works already on both platforms but I include it only for Linux because it is not necessary on Windows, it doesn't contain the definition for size_t, brings nothing useful.
Actually it brings STL behind the scene, in a file that don't use STL at all.